internal/service/wallet.go

```diff
@@ -660,6 +782,7 @@ func (s *walletService) ExpireCredits(ctx context.Context, transactionID string)
 		TransactionReason: types.TransactionReasonCreditExpired,
 		ReferenceType:     types.WalletTxReferenceTypeRequest,
 		ReferenceID:       tx.ID,
+		IdempotencyKey:    tx.IdempotencyKey,
```

Comment:
This will always fail as you are using the same Idempotency key will be same as original txn and this will fail in DB